libminifi/include/FlowController.h

```diff
@@ -338,6 +335,10 @@ class FlowController : public core::controller::ControllerServiceProvider, publi
   utils::optional<std::chrono::milliseconds> loadShutdownTimeoutFromConfiguration();
 
  private:
+  void restartThreadPool();
+  void initializeUninitializedSchedulers();
+  void reinitializeSchedulersWithNewThreadPool();
```

Comment:
I'm not sure if the name of this function actually reflect what it does. 
There is no new threadpool, as far as I see it's created only once and stopped/started according to usage later.

Reply:
It calls `ThreadPool::shutdown` which clears any running activities, joins all threads, and clears all worker threads as well (also resets config). The `ThreadPool` object is not new as in newly allocated, but new as the pool brand new threads.

I agree that it is not the best name, will rename it to `reinitializeSchedulersWithClearedThreadPool()`.